packages/@sfx/sayt-driver-plugin/src/sayt-driver-plugin.ts

```diff
@@ -33,19 +35,41 @@ export default class SaytDriverPlugin implements Plugin {
   /**
    * Event name to listen for Sayt autocomplete requests.
    */
-  saytDataEvent: string = 'sfx::autocomplete_fetch_data';
+  autocompleteRequestEvent: string = 'sfx::autocomplete_fetch_data';
   /**
    * Event name for sending Sayt autocomplete responses.
    */
-  saytResponseEvent: string = 'sfx::autocomplete_received_results';
+  autocompleteResponseEvent: string = 'sfx::autocomplete_received_results';
   /**
    * Event name to listen for Sayt autocomplete errors.
    */
-  saytErrorEvent: string = 'sfx::autocomplete_sayt_error';
+  autocompleteErrorEvent: string = 'sfx::autocomplete_sayt_error';
+  /**
+   * Event name to listen for Sayt product requests.
+   */
+  productRequestEvent: string = 'sfx::sayt_products_request';
+  /**
+   * Event name for sending Sayt product responses.
+   */
+  productResponseEvent: string = 'sfx::sayt_products_response';
+  /**
+   * Event name to listen for Sayt product errors.
+   */
+  productErrorEvent: string = 'sfx::sayt_products_error';
+  /**
+   * Provide default configuration for SAYT product searches.
+   */
+  defaultSearchConfig: object = {
+    fields: ['*'],
+  }
 
   constructor() {
-    this.fetchSaytData = this.fetchSaytData.bind(this);
+    this.fetchAutocompleteTerms = this.fetchAutocompleteTerms.bind(this);
+    this.fetchProductData = this.fetchProductData.bind(this);
+    this.sendAutocompleteApiRequest = this.sendAutocompleteApiRequest.bind(this);
```

Comment:
I'm of the mindset that a function needs to take care of its own interests. If a) this function is not usable without context, and b) we have no intention of purposely calling it with different contexts, we should bind it. It saves us in the future from bugs. I'm sure there are those that disagree with me.

Realistically, everything in a class should be automatically bound, but JavaScript wasn't written that way. Because we have to write it out method bindings manually, we are forced to choose which ones we bind. I'd prefer to choose not the minimal set for our current functionality, but a somewhat less minimal set that minimizes problems in the future.

Reply:
Yup, I'm fine if we want to bind everything as I don't think it's that costly. Although, I would think that if "b)" were true, then there's less reason to bind it.

Reply:
What I meant by (b) was that there could be cases where you want a method to act differently depending on where it's called. The method's context would change, which you might want in that particular edge case. This almost never happens.

Keep in mind that simply running the following will cause the context to change:
```js
const fetchTerms = driver.fetchAutocompleteTerms;
fetchTerms();
```
This means that, whether or not we expect a method to be called in different places, it is very easy to accidentally supply a different context, such as in a test. That's why I think it's important to do.

Reply:
My two problems with binding everything:

1. It doesn't allow for the context change at all, because a bound function cannot be rebound, and the default expectation to anyone who knows how `this` works is that a function is not bound.
2. It defeats the purpose of putting functions on the prototype because each copy of the component now gets its own set of functions. This can be costly, especially for components that have many instances.

Ideally, we would bind functions in the smallest necessary scope (i.e. at the place where it's passed down or assigned), but binding the function in the class constructor and putting it as an own property greatly simplifies testing and unregistering.

In my opinion, in the context of our components, we should only be binding functions that will be used as event listeners.

Reply:
@jmbtutor
After thinking about this, given that the Sayt driver exposes no methods, we only have to worry about binding the event listeners. More generally, though, I disagree with your reasons (1) and (2).

For (1), if you can think of a reason why someone might want to rebind the function to another context, I'd be open to changing it. I've almost never seen this done on purpose in the wild. One exception is binding `this` to functions that are in the current scope (but automatically define their own context). Another exception is binding event listeners.

(2) It's not surprising to me that an instance receives its own set of functions. If you feel that 'costly' here is significant, I am open to changing it - I don't need you spending the time finding benchmarks. Given that it's pretty easy to accidentally remove the scope of a function you want to use in JavaScript, the simplest thing to do is bind the functions you're worried about. Additionally, we do not expect many instances of any of our plugins or drivers.

For the record, I never suggested binding everything. I suggested binding any function that uses `this`, and that's my default stance for code safety and predictability. I've run into issues around context binding enough times that I write programs around that concern defensively.

Reply:
Agree to disagree on the finer points, but I think we agree from a practical standpoint, so I'm okay to resolve this.